### PR TITLE
AdEx Simplified Neuron Model

### PR DESCRIPTION
**Pull request by paulmyr**
Implementation mentioned in #774.

Notes:
I did not enable surrogate gradient support, because it is added already in v1.0.0.
I tried to directly adept it to v1.0.0 (also adapted to the channel API etc.), but I struggled to get the pytests working, due to old jaxley_mech version.
Therefore I stuck to the pattern of izhikevich.py which is currently available in the main branch.

To update to versions 1.0.0 surrogate gradient, one would have to replace
```python
        # Check for spike and reset
        spike_occurred = (
            v >= v_threshold
        )  # ToDo: replace this with heaviside function in v1.0.0
        v = jax.lax.select(spike_occurred, v_reset, v)
        w = jax.lax.select(spike_occurred, w + b, w)
```
with
```python
        spike_occurred = heaviside(voltage - v_threshold)
        v = spike_occurred * v_reset + (1.0 - spike_occurred) * v
        w = spike_occurred * (w + b) + (1.0 - spike_occurred) * w
```

[This google colab script](https://colab.research.google.com/github/paulmyr/DD2375-HT25-Project-Course-in-High-Performance-Computing/blob/master/5_evaluation/notebooks/jaxley_test_adex_version_pullrequest.ipynb) demonstrates the model using different parameters and compares them to brian2.

Like I said, this is my first pull request to an open source project.
If I did anything wrong or have to change it, if you have any other feedback otherwise, I greatly appreciate the feedback.

### CODE REVIEW

**Review by michaeldeistler — APPROVED** [review]
Hi Paul,

sorry for the silence, I had overlooked this after my vacation.

Anywyas: thanks a lot for this, it looks fantastic! Please add a line to the `CHANGELOG.md` to describe your changes. Otherwise I think this is good to go.

@jnsbck do you want to have a look?

Thanks again for the contribution, this is really cool!
Michael

**Comment by jnsbck** [comment]
lgtm! Do we want to merge this into `v1.0` rather than main and add https://github.com/paulmyr/jaxley/blob/ff612e7ea0c9d76655a9c4e7da8e6175a7b14a51/jaxley/channels/non_capacitive/adex.py#L151? @michaeldeistler 

On a side note, the changelog check passed, w.o. an update. Already know why. Will fix this.

**Comment by michaeldeistler** [comment]
Please use 0.14.0. And yes, tests on 1.0.0 are currently failing, don't worry about that :)
